### ISSUE TITLE
Application broker creates one service class for application, and plans per api-package

### ISSUE THREAD

**Issue by PK85** [kyma-project/kyma#7201]
**Description**
Application broker creates one service class for application, and plans per api-package.

**AC**
- make sure Application CR ID is a service class ID. Label that service class that UI will now is a new way how we represent applications in the Service Catalog.
- make sure Application#Service IDs are service plans IDs
- update existing documentation to reflect new design

**Related ticket**
- UI: https://github.com/kyma-project/console/issues/1544

**Comment by mszostok** [comment]
**Alternative plan:**

- Align contract with X-team about API Packages.
- Change Appilcation broker to be a cluster-wide and get rid of ApplicationMappings
- Remove controller for Application CR and replace that with graphql client to Director
- Change mapping and hardcoded things around single API and Events to support multi-entries
- Make sure that proxy implemented: https://github.com/kyma-incubator/compass/issues/752

**Comment by mszostok** [comment]
**New APP CR**

```
apiVersion: applicationconnector.kyma-project.io/v1alpha1
kind: Application
metadata:
  # {{ new requirement 
  #required, MUST only contain lowercase characters, numbers and hyphens (no spaces)
  # MUST be unique across all services. MUST be a non-empty string. !!!!!!!!
  # }}
  name: ec-prod
spec:
  accessLabel: access-label-1
  description: "EC description"

  # {{ move from service
  tags:
    - occ
    - promotions
  labels:
    connected-app: "ec-prod"
  # required, mapped to metadata.longDescription, used by catalog UI
  longDescription: "Long description for Promotions APIs"
  # mandatory, maps to metadata.providerDisplayName, provider name in catalog UI
  providerDisplayName: "HakunaMatata"
  # }}

  # {{ new fields
  displayName: "name"
  id: "foo-bar-123" # unique
  # }}

  services:
    # mandatory, unique ID
    - id: "ac031e8c-9aa4-4cb7-8999-0d358726ffaa"               # required, uuid, immutable
      # required, MUST only contain lowercase characters, numbers and hyphens (no spaces)
      # MUST be unique across all services. MUST be a non-empty string.
      name: "promotions-ac031e8c"
      # required, mapped to metadata.displayName, used by catalog UI
      displayName: "Promotions"
      # required, mapped to description, used by catalog UI
      description: "Short desc for Promotions APIs"

      # {{ new fields
      schemas:
        ServiceInstance: "JSON"
        ServiceBinding: "JSON"
      # }}

      # Entries defines, what can be enabled by activation (instantiating and/or binding) of the service
      entries:
      # type API defines remote API
      - type: API
        # the url, which can be used by service/lambda to make a call to remote API
        gatewayUrl: "http://promotions-gateway.production.svc.cluster.local/"+/secretName/dupa

        # {{ new fields
        name: "api-name"
        # }}

        # used in istio rules, generated by Wormhole, must be unique in k8s cluster
        # {{ not used anymore
        accessLabel: "access-label-1"
        # }}
      - type: API
        gatewayUrl: "http://promotions-gateway.production.svc.cluster.local/"
        name: "api-name"
      - type: API
        gatewayUrl: "http://promotions-gateway.production.svc.cluster.local/"
        name: "api-name"
      - type: Events
      - type: Events

    # second service provided by application
    - id: "48ab05bf-9aa4-4cb7-8999-0d3587265ac3"
      name: "orders-48ab05bf"
      displayName: "Orders"
      description: "Short desc for Orders API"
      entries:
      - type: API
        gatewayUrl: "http://orders-gateway.production.svc.cluster.local/"
        name: "api-name"
```

**Comment by klaudiagrz** [comment]
Estimations for both solutions:

Alternative plan without Application CR: ~ 40 SP
New APP CR: ~ 21 SP

--- 

Additionally, for both solutions we must decide on either:
- storage based on postgres (~ minimum 21 SP)
- storage based on configmap (~ 8 SP)

**Comment by mszostok** [comment]
alternative:
- without storage: when creating an auth add context info (e.g. instance ID) and just query the `package.instanceAuths()` and filter returned auths based on context. (5SP)

**Comment by mszostok** [comment]
Question:
**Generate the Auths per instance or per binding?**

**Comment by mszostok** [comment]
Here are all PRs:

- Add support to api packages in Application Broker and Application CRD #7330
- Update Application CRD Go type to support api-packages #7334
- Change logging middleware in App Broker server #7401
- Fix logging Kn Sub in App Broker #7407

Here is a task for removing old App Broker implementation after full support of API Packages in Kyma: https://github.com/kyma-project/kyma/issues/7415
